src/Wallabag/CoreBundle/Helper/ContentProxy.php

```diff
@@ -305,6 +312,90 @@ private function stockEntry(Entry $entry, array $content)
         }
     }
 
+    /**
+     * Update the origin_url field when a redirection occurs
+     * This field is set if it is empty and new url does not match ignore list.
+     *
+     * @param Entry  $entry
+     * @param string $url
+     */
+    private function updateOriginUrl(Entry $entry, $url)
+    {
+        if (!empty($url) && $entry->getUrl() !== $url) {
+            $parsed_entry_url = parse_url($entry->getUrl());
+            $parsed_content_url = parse_url($url);
+
+            $diff_ec = array_diff_assoc($parsed_entry_url, $parsed_content_url);
+            $diff_ce = array_diff_assoc($parsed_content_url, $parsed_entry_url);
+
+            $diff = array_merge($diff_ec, $diff_ce);
+            $diff_keys = array_keys($diff);
+            sort($diff_keys);
+
+            if ($this->ignoreUrl($entry->getUrl())) {
+                $entry->setUrl($url);
+            } else {
+                switch ($diff_keys) {
```

Comment:
Could you explain that `swtich`? It's a bit unclear what is expected here and also what the `case`s means.

Reply:
I'll add a comment to explain this part